src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -226,104 +226,68 @@ std::wstring_view Terminal::GetWorkingDirectory() noexcept
+    // GH#3494: We don't need to reflow the alt buffer. Apps that use the
+    // alt buffer will redraw themselves. This prevents graphical artifacts.
+    //
+    // This is consistent with VTE
     if (_inAltBuffer())
     {
-        // stash this resize for the future.
+        // _deferredResize will indicate to UseMainScreenBuffer() that it needs to reflow the main buffer.
+        // It's unknown why we defer the reflow to a later time. Performance perhaps?
         _deferredResize = viewportSize;
 
-        _altBuffer->GetCursor().StartDeferDrawing();
-        // we're capturing `this` here because when we exit, we want to EndDefer on the (newly created) active buffer.
-        auto endDefer = wil::scope_exit([this]() noexcept { _altBuffer->GetCursor().EndDeferDrawing(); });
+        _altBuffer->ResizeTraditional(viewportSize);
 
-        // GH#3494: We don't need to reflow the alt buffer. Apps that use the
-        // alt buffer will redraw themselves. This prevents graphical artifacts.
-        //
-        // This is consistent with VTE
-        RETURN_IF_FAILED(_altBuffer->ResizeTraditional(viewportSize));
-
-        // Since the _mutableViewport is no longer the size of the actual
-        // viewport, then update our _altBufferSize tracker we're using to help
-        // us out here.
         _altBufferSize = viewportSize;
+        _altBuffer->TriggerRedrawAll();
         return S_OK;
     }
 
-    const auto dx = viewportSize.width - oldDimensions.width;
-    const auto newBufferHeight = std::clamp(viewportSize.height + _scrollbackLines, 0, SHRT_MAX);
-
-    til::size bufferSize{ viewportSize.width, newBufferHeight };
-
-    // This will be used to determine where the viewport should be in the new buffer.
-    const auto oldViewportTop = _mutableViewport.Top();
-    auto newViewportTop = oldViewportTop;
-    auto newVisibleTop = _VisibleStartIndex();
+    const auto newBufferHeight = std::clamp(viewportSize.height + _scrollbackLines, 1, SHRT_MAX);
+    const til::size bufferSize{ viewportSize.width, newBufferHeight };
 
     // If the original buffer had _no_ scroll offset, then we should be at the
     // bottom in the new buffer as well. Track that case now.
     const auto originalOffsetWasZero = _scrollOffset == 0;
 
-    // skip any drawing updates that might occur until we swap _buffer with the new buffer or if we exit early.
-    _mainBuffer->GetCursor().StartDeferDrawing();
-    // we're capturing `this` here because when we exit, we want to EndDefer on the (newly created) active buffer.
-    auto endDefer = wil::scope_exit([this]() noexcept { _mainBuffer->GetCursor().EndDeferDrawing(); });
+    // GH#3848 - Stash away the current attributes the old text buffer is
+    // using. We'll initialize the new buffer with the default attributes,
+    // but after the resize, we'll want to make sure that the new buffer's
+    // current attributes (the ones used for printing new text) match the
+    // old buffer's.

```

Comment:
we no longer do this. is it because this linked bug is fixed in a different way? or because it's not necessary? update comment!

Reply:
ah, i see. it might have to move down to R290